### ISSUE TITLE
[SSR Module Federation - React] - Cannot read properties of undefined (reading 'target')

### ISSUE THREAD

**Issue by sahilpurav** [nrwl/nx#17246]
### Current Behavior

I'm using NX v16.2.2

As suggested in the [documentation](https://nx.dev/recipes/module-federation/module-federation-with-ssr),

- I'm running `npx nx g @nx/react:host host --ssr` command
- To serve the application, I ran `npx nx serve host`

**One I do this, I get following error:**

<img width="1347" alt="image" src="https://github.com/nrwl/nx/assets/8158506/28505d9a-7ac7-4d4d-a11b-4dbfe3af5ddb">

Please note that, when I run the application on http://localhost:4200, it runs properly. However, it doesn't watch for the file changes. This issue only exists in when we configure the host and remote app with "SSR" flag.

### Expected Behavior

NX should successfully serve the application and watch for the file changes.

### Steps to Reproduce

1. Configure the NX workspace to use "Integrated Monorepo"
2. Install @nx/react (npm i @nx/react)
3. Run npx nx g @nx/react:host store --ssr --remotes=product,checkout
4. Run npx nx serve store and you will notice the error.

### Nx Report

```shell
   Node   : 18.16.0
   OS     : darwin x64
   npm    : 9.5.1
   Hasher : Native

   nx                 : 16.2.2
   @nx/js             : 16.2.2
   @nx/linter         : 16.2.2
   @nx/workspace      : 16.2.2
   @nx/cypress        : 16.2.2
   @nx/devkit         : 16.2.2
   @nx/eslint-plugin  : 16.2.2
   @nx/react          : 16.2.2
   @nrwl/tao          : 16.2.2
   @nx/web            : 16.2.2
   @nx/webpack        : 16.2.2
   typescript         : 5.0.4
```

### Failure Logs

```shell
>  NX   Cannot read properties of undefined (reading 'target')


TypeError: Cannot read properties of undefined (reading 'target')
    at ssrDevServerExecutor_1 (/private/var/www/kolabverse/node_modules/@nx/webpack/src/executors/ssr-dev-server/ssr-dev-server.impl.js:31:40)
    at ssrDevServerExecutor_1.next (<anonymous>)
    at resume (/private/var/www/kolabverse/node_modules/tslib/tslib.js:279:48)
    at fulfill (/private/var/www/kolabverse/node_modules/tslib/tslib.js:281:35)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

 >  NX   Running target serve for project host failed

Error: write EPIPE                                                                                                                                         1 ✘  4s
    at target._send (node:internal/child_process:865:20)
    at target.send (node:internal/child_process:738:19)
    at /private/var/www/kolabverse/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:27:31
    at new Promise (<anonymous>)
    at sendMessage (/private/var/www/kolabverse/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:19:38)
    at /private/var/www/kolabverse/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:60:27
    at Generator.next (<anonymous>)
    at fulfilled (/private/var/www/kolabverse/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:5:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

This works when I don't use SSR flag. But I want my micro-frontends to be able to run on SSR

**Comment by sahilpurav** [comment]
@AgentEnder I have opened a PR which fixes this error. Ideally we should have received the "target" value but when I debugged the code, I haven't received any such variable.

**Comment by adrianiskandar** [comment]
This issue is closed and marked as solved I believe? But the issue still persists: I checked the related PR https://github.com/nrwl/nx/pull/17403 mentioning is fixed in version  `16.4.0-beta.2` but if I try the fix on the repo(package.json):

```
   "@nx/react": "16.4.0-beta.2",
    "@nx/web": "16.3.2",
    "@nx/webpack": "16.4.0-beta.2",
```

The issue still persists and now is a little worse I believe because it shows the node error for the host and the remote module(before it was just one).  It keeps one process running on 4200(the remote module) you have to kill it manually to try again(which is annoying). Additionally is not starting the apps in the correct ports, it should start on the port indicated in project.json(4201) instead. Because is complaining about `output.options` missing, I think maybe it makes sense? is not reading the options. Here are some more details of the error in version 16.4.0-beta.2

```
/Users/adrian/mod-fed-ssr/node_modules/@nx/react/src/executors/module-federation-ssr-dev-server/module-federation-ssr-dev-server.impl.js:48
                            PORT: remoteProject.targets['serve-browser'].options.port,
                                                                         ^

TypeError: Cannot read properties of undefined (reading 'options')
    at /Users/adrian/mod-fed-ssr/node_modules/@nx/react/src/executors/module-federation-ssr-dev-server/module-federation-ssr-dev-server.impl.js:48:74
    at Generator.next (<anonymous>)
    at /Users/adrian/mod-fed-ssr/node_modules/tslib/tslib.js:167:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/adrian/mod-fed-ssr/node_modules/tslib/tslib.js:163:16)
    at /Users/adrian/mod-fed-ssr/node_modules/@nx/react/src/executors/module-federation-ssr-dev-server/module-federation-ssr-dev-server.impl.js:42:128
    at [Symbol.asyncIterator] (/Users/adrian/mod-fed-ssr/node_modules/@nx/react/node_modules/@nx/devkit/src/utils/async-iterable/create-async-iterable.js:11:13)
    at f_1 (/Users/adrian/mod-fed-ssr/node_modules/@nx/react/node_modules/@nx/devkit/src/utils/async-iterable/map-async-iteratable.js:10:44)
    at f_1.next (<anonymous>)
    at resume (/Users/adrian/mod-fed-ssr/node_modules/tslib/tslib.js:279:48)

Node.js v20.2.0

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Running target serve for project react-host failed

   Failed tasks:
   
   - react-host:serve:development
   
   Hint: run the command with --verbose for more details.


 mod-fed-ssr on  main [!] via  v20.2.0 took 10s 
❯ Error: write EPIPE
    at target._send (node:internal/child_process:872:20)
    at target.send (node:internal/child_process:745:19)
    at /Users/adrian/mod-fed-ssr/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:27:31
    at new Promise (<anonymous>)
    at sendMessage (/Users/adrian/mod-fed-ssr/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:19:38)
    at /Users/adrian/mod-fed-ssr/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:60:27
    at Generator.next (<anonymous>)
    at fulfilled (/Users/adrian/mod-fed-ssr/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:5:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}
Error: write EPIPE
    at target._send (node:internal/child_process:872:20)
    at target.send (node:internal/child_process:745:19)
    at /Users/adrian/mod-fed-ssr/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:27:31
    at new Promise (<anonymous>)
    at sendMessage (/Users/adrian/mod-fed-ssr/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:19:38)
    at /Users/adrian/mod-fed-ssr/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:60:27
    at Generator.next (<anonymous>)
    at fulfilled (/Users/adrian/mod-fed-ssr/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/expose-rpc.js:5:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errno: -32,
  code: 'EPIPE',
```

**Comment by BorisTB** [comment]
I have the same issue in v16.5.3. Host is SSR and remote is not and it looks like host expects all remotes to be ssr as well
